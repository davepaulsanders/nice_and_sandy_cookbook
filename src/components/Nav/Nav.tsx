const Nav = ({
	isOpen,
	routes,
}: {
	isOpen: boolean;
	routes: { label: string; path: string }[];
}) => {
	return (
		<div
			className={`${isOpen ? "opacity-1 z-[1000]" : "opacity-0 z-[-1]"} ml-auto sticky border p-0 bg-darkg top-0 w-[20rem] text-lightg text-left pb-2 pt-6`}
		>
			<ul className="w-[20rem]">
				{routes.map((route) => (
					<a
						href={route.path}
						className={`block w-1/2 hover:text-lgg ${!isOpen ? "pointer-events-none" : ""}`}
						target="blank"
					>
						<li className="cursor-pointer pl-6 mb-8">
							{route.label}
						</li>
					</a>
				))}
			</ul>
		</div>
	);
};
export default Nav;
