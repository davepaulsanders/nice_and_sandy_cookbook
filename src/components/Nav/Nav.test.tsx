import Nav from "./Nav";
import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import "@testing-library/dom";
const routes = [{label: "Add a recipe", path: "/add"}, {label: "Complain", path: "https://github.com/davepaulsanders/nice_and_sandy_cookbook/issues"}]
test("<Nav /> renders open", () => {
	const { container } = render(
		<Nav routes={routes} isOpen={true} />
	);
	expect(container).toMatchSnapshot();
});
test("<Nav /> renders closed", () => {
	const { container } = render(
		<Nav routes={routes} isOpen={false} />
	);
	expect(container).toMatchSnapshot();
});
