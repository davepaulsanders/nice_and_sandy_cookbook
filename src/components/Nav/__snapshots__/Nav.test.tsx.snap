// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Nav /> renders closed 1`] = `
<div>
  <div
    class="opacity-0 ml-auto sticky border z-[1000] p-0 bg-darkg top-0 w-[20rem] text-lightg text-left pb-2 pt-6"
  >
    <ul
      class="w-[20rem]"
    >
      <a
        class="block w-1/2 hover:text-lgg pointer-events-none"
        href="/add"
        target="blank"
      >
        <li
          class="cursor-pointer pl-6 mb-8"
        >
          Add a recipe
        </li>
      </a>
      <a
        class="block w-1/2 hover:text-lgg pointer-events-none"
        href="https://github.com/davepaulsanders/nice_and_sandy_cookbook/issues"
        target="blank"
      >
        <li
          class="cursor-pointer pl-6 mb-8"
        >
          Complain
        </li>
      </a>
    </ul>
  </div>
</div>
`;

exports[`<Nav /> renders open 1`] = `
<div>
  <div
    class="opacity-1 ml-auto sticky border z-[1000] p-0 bg-darkg top-0 w-[20rem] text-lightg text-left pb-2 pt-6"
  >
    <ul
      class="w-[20rem]"
    >
      <a
        class="block w-1/2 hover:text-lgg "
        href="/add"
        target="blank"
      >
        <li
          class="cursor-pointer pl-6 mb-8"
        >
          Add a recipe
        </li>
      </a>
      <a
        class="block w-1/2 hover:text-lgg "
        href="https://github.com/davepaulsanders/nice_and_sandy_cookbook/issues"
        target="blank"
      >
        <li
          class="cursor-pointer pl-6 mb-8"
        >
          Complain
        </li>
      </a>
    </ul>
  </div>
</div>
`;
